hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/KeyOutputStream.java

```diff
@@ -555,6 +565,12 @@ public synchronized void close() throws IOException {
       if (!isException) {
         Preconditions.checkArgument(writeOffset == offset);
       }
+      if (requiresAtomicWrite) {
+        long expectedSize = blockOutputStreamEntryPool.getDataSize();
+        Preconditions.checkArgument(expectedSize == offset,
```

Comment:
Suggest to use Preconditions.checkState here, so it will throw out IllegalStateException instead of
IllegalArgumentException.  IllegalStateException is more appropriate than IllegalArgumentException here.

Reply:
And please catch this exception in ObjectEndpoint#put  finally when OutputStream#close is called.

Reply:
Done.